File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/py_to_cpp/ComplexCalculator.cpp
#include <complex>

class ComplexCalculator {
public:
    static std::complex<double> add(const std::complex<double>& c1, const std::complex<double>& c2) {
        return c1 + c2;
    }

    static std::complex<double> subtract(const std::complex<double>& c1, const std::complex<double>& c2) {
        return c1 - c2;
    }

    static std::complex<double> multiply(const std::complex<double>& c1, const std::complex<double>& c2) {
        return c1 * c2;
    }

    static std::complex<double> divide(const std::complex<double>& c1, const std::complex<double>& c2) {
        return c1 / c2;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/cpp_to_py/code_output/ComplexCalculator.py
class ComplexCalculator:
    @staticmethod
    def add(c1, c2):
        return complex(c1.real + c2.real, c1.imag + c2.imag)

    @staticmethod
    def subtract(c1, c2):
        return complex(c1.real - c2.real, c1.imag - c2.imag)

    @staticmethod
    def multiply(c1, c2):
        return complex(c1.real * c2.real - c1.imag * c2.imag, c1.real * c2.imag + c1.imag * c2.real)

    @staticmethod
    def divide(c1, c2):
        denominator = c2.real * c2.real + c2.imag * c2.imag
        real = (c1.real * c2.real + c1.imag * c2.imag) / denominator
        imaginary = (c1.imag * c2.real - c1.real * c2.imag) / denominator
        return complex(real, imaginary)